vars/phaseDeploy.groovy

```diff
@@ -13,18 +13,41 @@ def call(Project project, List<Set<Map>> repos) {
 
     def phase = MROPipelineUtil.PipelinePhases.DEPLOY
 
+    def standardWorkspace = env.WORKSPACE
+    def agentPodCondition = project.isPromotionMode
+
     def preExecuteRepo = { steps, repo ->
-        levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_EXECUTE_REPO, repo)
+        // In case we run the phase on an agent node, we need to make sure that
+        // the levaDocScheduler.run is executed on the master node, as it does
+        // not work on agent nodes yet.
+        if (agentPodCondition) {
+            node {
+                sh "cp -r ${standardWorkspace}/docs ${env.WORKSPACE}/docs"
+                levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_EXECUTE_REPO, repo)
+            }
+        } else {
+            levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_EXECUTE_REPO, repo)
+        }
     }
 
     def postExecuteRepo = { steps, repo ->
-        levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.POST_EXECUTE_REPO, repo, repo.data)
+        // In case we run the phase on an agent node, we need to make sure that
+        // the levaDocScheduler.run is executed on the master node, as it does
+        // not work on agent nodes yet.
+        if (agentPodCondition) {
+            node {
+                sh "cp -r ${standardWorkspace}/docs ${env.WORKSPACE}/docs"
+                levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.POST_EXECUTE_REPO, repo, repo.data)
+            }
+        } else {
+            levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_EXECUTE_REPO, repo)
```

Comment:
Shouldn't this be `POST_EXECUTE_REPO` and contain `repo.data` ? 
And this agent pod condition here and in line 23 looks like a true duplication, right? Could it be extracted-out into a method named something like `runScedulerOnMasterNode`

Reply:
@michaelsauter  - that's a true bug ...  :)

Reply:
@martsec Great catch! Resolved in 0f5bc4b.